### PR TITLE
8279488: ProcessBuilder inherits contextClassLoader when spawning a process reaper thread

### PR DESCRIPTION
**Pull request by RogerRiggs**
The thread factory used to create the process reaper threads unnecessarily inherits the callers thread context classloader.
The result is retention of the class loader.

The thread factory used for the pool of process reaper threads is modified to use an InnocuousThread with a given stacksize.
The test verifies that the process reaper threads have a null context classloader.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8279488](https://bugs.openjdk.java.net/browse/JDK-8279488): ProcessBuilder inherits contextClassLoader when spawning a process reaper thread

### Reviewers
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7131/head:pull/7131` \
`$ git checkout pull/7131`

Update a local copy of the PR: \
`$ git checkout pull/7131` \
`$ git pull https://git.openjdk.java.net/jdk pull/7131/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7131`

View PR using the GUI difftool: \
`$ git pr show -t 7131`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7131.diff">https://git.openjdk.java.net/jdk/pull/7131.diff</a>

</details>

### CODE REVIEW

**Comment by RogerRiggs** [comment]
@AlanBateman Can you take a look at this use of InnocuousThread.  Thanks

**Comment by RogerRiggs** [comment]
Please Review.

File: src/java.base/share/classes/jdk/internal/misc/InnocuousThread.java [hunk 1]
```diff
@@ -104,28 +104,46 @@ public static Thread newSystemThread(String name, Runnable target) {
...
+                        return createThread(name, target, 0L, null, priority);
+                    }
+                });
+    }
+
+    private static Thread createThread(String name, Runnable target, long stackSize, ClassLoader loader, int priority) {
         Thread t = new InnocuousThread(INNOCUOUSTHREADGROUP,
-                target, name, loader);
+                target, name, stackSize, loader);
         if (priority >= 0) {
```
Comment (AlanBateman) [inline src/java.base/share/classes/jdk/internal/misc/InnocuousThread.java:138]: Adding an addition newSystemThread method should be okay but can you fix the formatting/line lengths so it's a bit more inconsistent with the existing code.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**